Script >64KB leads to Could not open cache directory:  java.lang.IllegalArgumentException

<!--- Provide a brief summary of the issue in the title above -->
Project Configuration fails when the build.gradle.kts is large.

### Expected Behavior
<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->
Being able to write an arbitrarily large build script.

### Current Behavior
<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If suggesting a change/improvement, explain the difference from current behavior -->
When there are more than approximately 65K characters in my build.gradle.kts script, project configuration fails. This doesn't appear if I remove a couple of Gradle tasks to decrease the number of characters.
```
 gradlew.bat compileJava --stacktrace

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'project_name'.
> Could not open cache directory 5xdkp5yubb79383gig042ae0w (%HOME%\.gradle\caches\5.2-20181230000028+0000\gradle-kotlin-dsl\5xdkp5yubb79383gig042ae0w).
   > java.lang.IllegalArgumentException (no error message)

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.ProjectConfigurationException: A problem occurred configuring root project 'project_name'.
        at org.gradle.configuration.project.LifecycleProjectEvaluator.wrapException(LifecycleProjectEvaluator.java:79)
        at org.gradle.configuration.project.LifecycleProjectEvaluator.addConfigurationFailure(LifecycleProjectEvaluator.java:72)
        at org.gradle.configuration.project.LifecycleProjectEvaluator.access$600(LifecycleProjectEvaluator.java:53)
        at org.gradle.configuration.project.LifecycleProjectEvaluator$EvaluateProject$1.run(LifecycleProjectEvaluator.java:108)
        at org.gradle.internal.Factories$1.create(Factories.java:25)
        at org.gradle.internal.work.DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:183)
        at org.gradle.internal.work.StopShieldingWorkerLeaseService.withLocks(StopShieldingWorkerLeaseService.java:40)
        at org.gradle.api.internal.project.DefaultProjectStateRegistry$ProjectStateImpl.withProjectLock(DefaultProjectStateRegistry.java:226)
        at org.gradle.api.internal.project.DefaultProjectStateRegistry$ProjectStateImpl.withMutableState(DefaultProjectStateRegistry.java:220)
        at org.gradle.api.internal.project.DefaultProjectStateRegistry$ProjectStateImpl.withMutableState(DefaultProjectStateRegistry.java:186)
        at org.gradle.configuration.project.LifecycleProjectEvaluator$EvaluateProject.run(LifecycleProjectEvaluator.java:95)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:402)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:394)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:92)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.configuration.project.LifecycleProjectEvaluator.evaluate(LifecycleProjectEvaluator.java:67)
        at org.gradle.api.internal.project.DefaultProject.evaluate(DefaultProject.java:695)
        at org.gradle.api.internal.project.DefaultProject.evaluate(DefaultProject.java:143)
        at org.gradle.execution.TaskPathProjectEvaluator.configure(TaskPathProjectEvaluator.java:35)
        at org.gradle.execution.TaskPathProjectEvaluator.configureHierarchy(TaskPathProjectEvaluator.java:60)
        at org.gradle.configuration.DefaultBuildConfigurer.configure(DefaultBuildConfigurer.java:41)
        at org.gradle.initialization.DefaultGradleLauncher$ConfigureBuild.run(DefaultGradleLauncher.java:302)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:402)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:394)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:92)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.initialization.DefaultGradleLauncher.configureBuild(DefaultGradleLauncher.java:210)
        at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:151)
        at org.gradle.initialization.DefaultGradleLauncher.executeTasks(DefaultGradleLauncher.java:134)
        at org.gradle.internal.invocation.GradleBuildController$1.execute(GradleBuildController.java:58)
        at org.gradle.internal.invocation.GradleBuildController$1.execute(GradleBuildController.java:55)
        at org.gradle.internal.invocation.GradleBuildController$3.create(GradleBuildController.java:82)
        at org.gradle.internal.invocation.GradleBuildController$3.create(GradleBuildController.java:75)
        at org.gradle.internal.work.DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:183)
        at org.gradle.internal.work.StopShieldingWorkerLeaseService.withLocks(StopShieldingWorkerLeaseService.java:40)
        at org.gradle.internal.invocation.GradleBuildController.doBuild(GradleBuildController.java:75)
        at org.gradle.internal.invocation.GradleBuildController.run(GradleBuildController.java:55)
        at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:31)
        at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
        at org.gradle.launcher.exec.BuildOutcomeReportingBuildActionRunner.run(BuildOutcomeReportingBuildActionRunner.java:58)
        at org.gradle.tooling.internal.provider.ValidatingBuildActionRunner.run(ValidatingBuildActionRunner.java:32)
        at org.gradle.launcher.exec.BuildCompletionNotifyingBuildActionRunner.run(BuildCompletionNotifyingBuildActionRunner.java:39)
        at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner$3.call(RunAsBuildOperationBuildActionRunner.java:49)
        at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner$3.call(RunAsBuildOperationBuildActionRunner.java:44)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:416)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:406)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:102)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
        at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner.run(RunAsBuildOperationBuildActionRunner.java:44)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter$1.transform(InProcessBuildActionExecuter.java:49)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter$1.transform(InProcessBuildActionExecuter.java:46)
        at org.gradle.composite.internal.DefaultRootBuildState.run(DefaultRootBuildState.java:78)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:46)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:31)
        at org.gradle.launcher.exec.BuildTreeScopeBuildActionExecuter.execute(BuildTreeScopeBuildActionExecuter.java:42)
        at org.gradle.launcher.exec.BuildTreeScopeBuildActionExecuter.execute(BuildTreeScopeBuildActionExecuter.java:28)
        at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:78)
        at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:52)
        at org.gradle.tooling.internal.provider.SubscribableBuildActionExecuter.execute(SubscribableBuildActionExecuter.java:59)
        at org.gradle.tooling.internal.provider.SubscribableBuildActionExecuter.execute(SubscribableBuildActionExecuter.java:36)
        at org.gradle.tooling.internal.provider.SessionScopeBuildActionExecuter.execute(SessionScopeBuildActionExecuter.java:68)
        at org.gradle.tooling.internal.provider.SessionScopeBuildActionExecuter.execute(SessionScopeBuildActionExecuter.java:38)
        at org.gradle.tooling.internal.provider.GradleThreadBuildActionExecuter.execute(GradleThreadBuildActionExecuter.java:37)
        at org.gradle.tooling.internal.provider.GradleThreadBuildActionExecuter.execute(GradleThreadBuildActionExecuter.java:26)
        at org.gradle.tooling.internal.provider.ParallelismConfigurationBuildActionExecuter.execute(ParallelismConfigurationBuildActionExecuter.java:43)
        at org.gradle.tooling.internal.provider.ParallelismConfigurationBuildActionExecuter.execute(ParallelismConfigurationBuildActionExecuter.java:29)
        at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:60)
        at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:32)
        at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:55)
        at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:41)
        at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:48)
        at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:32)
        at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:67)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:37)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:26)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:34)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:74)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:72)
        at org.gradle.util.Swapper.swap(Swapper.java:38)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:72)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.LogAndCheckHealth.execute(LogAndCheckHealth.java:55)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:62)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:81)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:50)
        at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:295)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: org.gradle.cache.CacheOpenException: Could not open cache directory 5xdkp5yubb79383gig042ae0w (%HOME%\.gradle\caches\5.2-20181230000028+0000\gradle-kotlin-dsl\5xdkp5yubb79383gig042ae0w).
        at org.gradle.cache.internal.DefaultPersistentDirectoryStore.open(DefaultPersistentDirectoryStore.java:80)
        at org.gradle.cache.internal.DefaultPersistentDirectoryStore.open(DefaultPersistentDirectoryStore.java:42)
        at org.gradle.cache.internal.DefaultCacheFactory.doOpen(DefaultCacheFactory.java:94)
        at org.gradle.cache.internal.DefaultCacheFactory.open(DefaultCacheFactory.java:68)
        at org.gradle.cache.internal.DefaultCacheRepository$PersistentCacheBuilder.open(DefaultCacheRepository.java:118)
        at org.gradle.kotlin.dsl.cache.ScriptCache.cacheDirFor(ScriptCache.kt:67)
        at org.gradle.kotlin.dsl.provider.StandardKotlinScriptEvaluator$InterpreterHost.cacheDirFor(KotlinScriptEvaluator.kt:253)
        at org.gradle.kotlin.dsl.provider.StandardKotlinScriptEvaluator$InterpreterHost.cachedDirFor(KotlinScriptEvaluator.kt:242)
        at org.gradle.kotlin.dsl.execution.Interpreter.emitSpecializedProgramFor(Interpreter.kt:260)
        at org.gradle.kotlin.dsl.execution.Interpreter.eval(Interpreter.kt:187)
        at org.gradle.kotlin.dsl.provider.StandardKotlinScriptEvaluator.evaluate(KotlinScriptEvaluator.kt:114)
        at org.gradle.kotlin.dsl.provider.KotlinScriptPluginFactory$create$1.invoke(KotlinScriptPluginFactory.kt:51)
        at org.gradle.kotlin.dsl.provider.KotlinScriptPluginFactory$create$1.invoke(KotlinScriptPluginFactory.kt:36)
        at org.gradle.kotlin.dsl.provider.KotlinScriptPlugin.apply(KotlinScriptPlugin.kt:34)
        at org.gradle.configuration.BuildOperationScriptPlugin$1$1.run(BuildOperationScriptPlugin.java:69)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:402)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:394)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:92)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.configuration.BuildOperationScriptPlugin$1.execute(BuildOperationScriptPlugin.java:66)
        at org.gradle.configuration.BuildOperationScriptPlugin$1.execute(BuildOperationScriptPlugin.java:63)
        at org.gradle.configuration.internal.DefaultUserCodeApplicationContext.apply(DefaultUserCodeApplicationContext.java:48)
        at org.gradle.configuration.BuildOperationScriptPlugin.apply(BuildOperationScriptPlugin.java:63)
        at org.gradle.configuration.project.BuildScriptProcessor$1.run(BuildScriptProcessor.java:44)
        at org.gradle.internal.Factories$1.create(Factories.java:25)
        at org.gradle.api.internal.project.DefaultProjectStateRegistry$ProjectStateImpl.withMutableState(DefaultProjectStateRegistry.java:200)
        at org.gradle.api.internal.project.DefaultProjectStateRegistry$ProjectStateImpl.withMutableState(DefaultProjectStateRegistry.java:186)
        at org.gradle.configuration.project.BuildScriptProcessor.execute(BuildScriptProcessor.java:41)
        at org.gradle.configuration.project.BuildScriptProcessor.execute(BuildScriptProcessor.java:26)
        at org.gradle.configuration.project.ConfigureActionsProjectEvaluator.evaluate(ConfigureActionsProjectEvaluator.java:34)
        at org.gradle.configuration.project.LifecycleProjectEvaluator$EvaluateProject$1.run(LifecycleProjectEvaluator.java:106)
        ... 104 more
Caused by: java.lang.IllegalArgumentException
        at org.jetbrains.org.objectweb.asm.ByteVector.putUTF8(ByteVector.java:213)
        at org.jetbrains.org.objectweb.asm.ClassWriter.newUTF8(ClassWriter.java:1141)
        at org.jetbrains.org.objectweb.asm.ClassWriter.newStringishItem(ClassWriter.java:1163)
        at org.jetbrains.org.objectweb.asm.ClassWriter.newConstItem(ClassWriter.java:1091)
        at org.jetbrains.org.objectweb.asm.MethodWriter.visitLdcInsn(MethodWriter.java:1165)
        at org.gradle.kotlin.dsl.support.bytecode.AsmExtensionsKt.LDC(AsmExtensions.kt:163)
        at org.gradle.kotlin.dsl.execution.ResidualProgramCompiler.emitCompileSecondStageScript(ResidualProgramCompiler.kt:357)
        at org.gradle.kotlin.dsl.execution.ResidualProgramCompiler.access$emitCompileSecondStageScript(ResidualProgramCompiler.kt:93)
        at org.gradle.kotlin.dsl.execution.ResidualProgramCompiler$overrideLoadSecondStageFor$1.invoke(ResidualProgramCompiler.kt:349)
        at org.gradle.kotlin.dsl.execution.ResidualProgramCompiler$overrideLoadSecondStageFor$1.invoke(ResidualProgramCompiler.kt:93)
        at org.gradle.kotlin.dsl.support.bytecode.AsmExtensionsKt.method(AsmExtensions.kt:117)
        at org.gradle.kotlin.dsl.support.bytecode.AsmExtensionsKt.publicMethod(AsmExtensions.kt:102)
        at org.gradle.kotlin.dsl.support.bytecode.AsmExtensionsKt.publicMethod$default(AsmExtensions.kt:99)
        at org.gradle.kotlin.dsl.execution.ResidualProgramCompiler.overrideLoadSecondStageFor(ResidualProgramCompiler.kt:331)
        at org.gradle.kotlin.dsl.execution.ResidualProgramCompiler.access$overrideLoadSecondStageFor(ResidualProgramCompiler.kt:93)
        at org.gradle.kotlin.dsl.execution.ResidualProgramCompiler$emitDynamicProgram$1.invoke(ResidualProgramCompiler.kt:132)
        at org.gradle.kotlin.dsl.execution.ResidualProgramCompiler$emitDynamicProgram$1.invoke(ResidualProgramCompiler.kt:93)
        at org.gradle.kotlin.dsl.execution.ResidualProgramCompiler$program$3.invoke(ResidualProgramCompiler.kt:523)
        at org.gradle.kotlin.dsl.execution.ResidualProgramCompiler$program$3.invoke(ResidualProgramCompiler.kt:93)
        at org.gradle.kotlin.dsl.support.bytecode.AsmExtensionsKt.publicClass-LgnzLRE(AsmExtensions.kt:37)
        at org.gradle.kotlin.dsl.execution.ResidualProgramCompiler.program-7MEG69U(ResidualProgramCompiler.kt:521)
        at org.gradle.kotlin.dsl.execution.ResidualProgramCompiler.access$program-7MEG69U(ResidualProgramCompiler.kt:93)
        at org.gradle.kotlin.dsl.execution.ResidualProgramCompiler.emitDynamicProgram(ResidualProgramCompiler.kt:638)
        at org.gradle.kotlin.dsl.execution.ResidualProgramCompiler.compile(ResidualProgramCompiler.kt:107)
        at org.gradle.kotlin.dsl.execution.Interpreter$emitSpecializedProgramFor$cachedDir$1.invoke(Interpreter.kt:296)
        at org.gradle.kotlin.dsl.execution.Interpreter$emitSpecializedProgramFor$cachedDir$1.invoke(Interpreter.kt:72)
        at org.gradle.kotlin.dsl.cache.ScriptCache.initializeCacheDir(ScriptCache.kt:114)
        at org.gradle.kotlin.dsl.cache.ScriptCache.access$initializeCacheDir(ScriptCache.kt:37)
        at org.gradle.kotlin.dsl.cache.ScriptCache$cacheDirFor$1.execute(ScriptCache.kt:60)
        at org.gradle.kotlin.dsl.cache.ScriptCache$cacheDirFor$1.execute(ScriptCache.kt:37)
        at org.gradle.cache.internal.DefaultPersistentDirectoryCache$Initializer.initialize(DefaultPersistentDirectoryCache.java:98)
        at org.gradle.cache.internal.FixedSharedModeCrossProcessCacheAccess$1.run(FixedSharedModeCrossProcessCacheAccess.java:85)
        at org.gradle.cache.internal.DefaultFileLockManager$DefaultFileLock.doWriteAction(DefaultFileLockManager.java:207)
        at org.gradle.cache.internal.DefaultFileLockManager$DefaultFileLock.writeFile(DefaultFileLockManager.java:197)
        at org.gradle.cache.internal.FixedSharedModeCrossProcessCacheAccess.open(FixedSharedModeCrossProcessCacheAccess.java:83)
        at org.gradle.cache.internal.DefaultCacheAccess.open(DefaultCacheAccess.java:142)
        at org.gradle.cache.internal.DefaultPersistentDirectoryStore.open(DefaultPersistentDirectoryStore.java:78)
        ... 137 more


* Get more help at https://help.gradle.org

BUILD FAILED in 2s
```

### Context
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->
* I'm trying to migrate a large monolithic project from Ant to Gradle. 
* I'm not using any Ant targets in my Gradle build script.
* When the Gradle Kotlin Built Script becomes sufficiently large, project configuration keeps failing.

### Steps to Reproduce (for bugs)
<!--- Provide a link to a build scan, example project, or an unambiguous -->
<!--- set of steps to reproduce this bug. Include code to reproduce, if relevant -->
*  I made a sample project, with a build script longer than 65K character, here is the link [GitHub Link](https://github.com/bhanutejags/sampleproject).
* Running `gradlew.bat`, will give throw `java.lang.IllegalArgumentException`.

### Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
<!--- A build scan `https://gradle.com/scans/get-started` is ideal -->
 * `gradle --version`
```
------------------------------------------------------------
Gradle 5.2-20181230000028+0000
------------------------------------------------------------

Build time:   2018-12-30 00:00:28 UTC
Revision:     5556139b12cca73f6e4e36d3c2d61b7d9fda159c

Kotlin DSL:   1.1.0
Kotlin:       1.3.11
Groovy:       2.5.4
Ant:          Apache Ant(TM) version 1.9.13 compiled on July 10 2018
JVM:          1.8.0_192 (Oracle Corporation 25.192-b12)
OS:           Windows 10 10.0 amd64
```

 * Version of IntelliJ or Android Studio (in `About` menu you can copy version information)
```
IntelliJ IDEA 2018.3.3 (Ultimate Edition)
Build #IU-183.5153.8, built on December 24, 2018
JRE: 1.8.0_152-release-1343-b26 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0
```

--- a/subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/cache/ScriptCache.kt
+++ b/subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/cache/ScriptCache.kt
@@ -71,7 +71,7 @@ class ScriptCache(
     }
 
     private
-    val cacheProperties = mapOf("version" to "14")
+    val cacheProperties = mapOf("version" to "15")
 
     private
     fun cacheDirOf(baseDir: File) = File(baseDir, "cache")

--- a/subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/execution/ExecutableProgram.kt
+++ b/subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/execution/ExecutableProgram.kt
@@ -22,6 +22,7 @@ import org.gradle.internal.classpath.ClassPath
 import org.gradle.internal.hash.HashCode
 
 import org.gradle.kotlin.dsl.support.KotlinScriptHost
+import org.gradle.kotlin.dsl.support.useToRun
 
 import org.gradle.plugin.management.internal.PluginRequests
 
@@ -69,8 +70,8 @@ abstract class ExecutableProgram {
             accessorsClassPath: ClassPath?
         )
 
-        fun compileSecondStageScript(
-            scriptText: String,
+        fun compileSecondStageOf(
+            program: StagedProgram,
             scriptHost: KotlinScriptHost<*>,
             scriptTemplateId: String,
             sourceHash: HashCode,
@@ -88,6 +89,8 @@ abstract class ExecutableProgram {
 
     abstract class StagedProgram : ExecutableProgram() {
 
+        abstract val secondStageScriptText: String
+
         abstract fun loadSecondStageFor(
             programHost: Host,
             scriptHost: KotlinScriptHost<*>,
@@ -95,5 +98,10 @@ abstract class ExecutableProgram {
             sourceHash: HashCode,
             accessorsClassPath: ClassPath?
         ): Class<*>
+
+        fun loadScriptResource(resourcePath: String): String =
+            javaClass.getResourceAsStream(resourcePath).bufferedReader().useToRun {
+                readText()
+            }
     }
 }

--- a/subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/execution/Interpreter.kt
+++ b/subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/execution/Interpreter.kt
@@ -426,8 +426,8 @@ class Interpreter(val host: Host) {
                 host.compilationClassPathOf(scriptHost.targetScope)
             ).bin
 
-        override fun compileSecondStageScript(
-            scriptText: String,
+        override fun compileSecondStageOf(
+            program: ExecutableProgram.StagedProgram,
             scriptHost: KotlinScriptHost<*>,
             scriptTemplateId: String,
             sourceHash: HashCode,
@@ -469,7 +469,7 @@ class Interpreter(val host: Host) {
 
                         scriptSource.withLocationAwareExceptionHandling {
 
-                            withTemporaryScriptFileFor(originalScriptPath, scriptText) { scriptFile ->
+                            withTemporaryScriptFileFor(originalScriptPath, program.secondStageScriptText) { scriptFile ->
 
                                 ResidualProgramCompiler(
                                     outputDir,

--- a/subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/execution/ResidualProgramCompiler.kt
+++ b/subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/execution/ResidualProgramCompiler.kt
@@ -20,6 +20,7 @@ import org.gradle.api.Project
 
 import org.gradle.internal.classpath.ClassPath
 import org.gradle.internal.hash.HashCode
+import org.gradle.internal.hash.Hashing
 
 import org.gradle.kotlin.dsl.KotlinBuildScript
 import org.gradle.kotlin.dsl.KotlinInitScript
@@ -129,7 +130,49 @@ class ResidualProgramCompiler(
                 emitEvaluateSecondStageOf()
             }
 
-            overrideLoadSecondStageFor(program.source.text)
+            overrideGetSecondStageScriptText(program.source.text)
+            overrideLoadSecondStageFor()
+        }
+    }
+
+    private
+    fun ClassWriter.overrideGetSecondStageScriptText(secondStageScriptText: String) {
+        publicMethod(
+            "getSecondStageScriptText",
+            "()Ljava/lang/String;",
+            "()Ljava/lang/String;"
+        ) {
+            if (secondStageScriptText.length < 16 * 1024) {
+                LDC(secondStageScriptText)
+            } else {
+                // Large scripts are stored as a resource to overcome
+                // the 64KB string constant limitation
+                val resourcePath = storeStringToResource(secondStageScriptText)
+                ALOAD(0)
+                LDC(resourcePath)
+                INVOKEVIRTUAL(
+                    ExecutableProgram.StagedProgram::class.internalName,
+                    ExecutableProgram.StagedProgram::loadScriptResource.name,
+                    "(Ljava/lang/String;)Ljava/lang/String;"
+                )
+            }
+            ARETURN()
+        }
+    }
+
+    private
+    fun storeStringToResource(secondStageScriptText: String): String {
+        val hash = Hashing.hashString(secondStageScriptText)
+        val resourcePath = "gradle-kotlin-dsl/$hash"
+        writeResourceFile(resourcePath, secondStageScriptText)
+        return resourcePath
+    }
+
+    private
+    fun writeResourceFile(resourcePath: String, resourceText: String) {
+        outputFile(resourcePath).apply {
+            parentFile.mkdir()
+            writeText(resourceText)
         }
     }
 
@@ -327,17 +370,17 @@ class ResidualProgramCompiler(
     }
 
     private
-    fun ClassWriter.overrideLoadSecondStageFor(scriptText: String) {
+    fun ClassWriter.overrideLoadSecondStageFor() {
         publicMethod(
-            "loadSecondStageFor",
-            "(" +
+            name = "loadSecondStageFor",
+            desc = "(" +
                 "Lorg/gradle/kotlin/dsl/execution/ExecutableProgram\$Host;" +
                 "Lorg/gradle/kotlin/dsl/support/KotlinScriptHost;" +
                 "Ljava/lang/String;" +
                 "Lorg/gradle/internal/hash/HashCode;" +
                 "Lorg/gradle/internal/classpath/ClassPath;" +
                 ")Ljava/lang/Class;",
-            "(" +
+            signature = "(" +
                 "Lorg/gradle/kotlin/dsl/execution/ExecutableProgram\$Host;" +
                 "Lorg/gradle/kotlin/dsl/support/KotlinScriptHost<*>;" +
                 "Ljava/lang/String;" +
@@ -346,31 +389,27 @@ class ResidualProgramCompiler(
                 ")Ljava/lang/Class<*>;"
         ) {
 
-            emitCompileSecondStageScript(scriptText)
+            ALOAD(Vars.ProgramHost)
+            ALOAD(0)
+            ALOAD(Vars.ScriptHost)
+            ALOAD(3)
+            ALOAD(4)
+            GETSTATIC(programKind)
+            GETSTATIC(programTarget)
+            ALOAD(5)
+            invokeHost(
+                ExecutableProgram.Host::compileSecondStageOf.name,
+                "(" +
+                    stagedProgramType +
+                    "Lorg/gradle/kotlin/dsl/support/KotlinScriptHost;" +
+                    "Ljava/lang/String;Lorg/gradle/internal/hash/HashCode;" +
+                    "Lorg/gradle/kotlin/dsl/execution/ProgramKind;" +
+                    "Lorg/gradle/kotlin/dsl/execution/ProgramTarget;" +
+                    "Lorg/gradle/internal/classpath/ClassPath;" +
+                    ")Ljava/lang/Class;"
+            )
             ARETURN()
         }
-    }
-
-    private
-    fun MethodVisitor.emitCompileSecondStageScript(scriptText: String) {
-        ALOAD(Vars.ProgramHost)
-        LDC(scriptText)
-        ALOAD(Vars.ScriptHost)
-        ALOAD(3)
-        ALOAD(4)
-        GETSTATIC(programKind)
-        GETSTATIC(programTarget)
-        ALOAD(5) // accessorsClassPath
-        invokeHost(
-            ExecutableProgram.Host::compileSecondStageScript.name,
-            "(" +
-                "Ljava/lang/String;" +
-                "Lorg/gradle/kotlin/dsl/support/KotlinScriptHost;" +
-                "Ljava/lang/String;Lorg/gradle/internal/hash/HashCode;" +
-                "Lorg/gradle/kotlin/dsl/execution/ProgramKind;" +
-                "Lorg/gradle/kotlin/dsl/execution/ProgramTarget;" +
-                "Lorg/gradle/internal/classpath/ClassPath;" +
-                ")Ljava/lang/Class;")
     }
 
     private
@@ -386,13 +425,16 @@ class ResidualProgramCompiler(
         invokeHost(
             ExecutableProgram.Host::evaluateSecondStageOf.name,
             "(" +
-                "Lorg/gradle/kotlin/dsl/execution/ExecutableProgram\$StagedProgram;" +
+                stagedProgramType +
                 "Lorg/gradle/kotlin/dsl/support/KotlinScriptHost;" +
                 "Ljava/lang/String;" +
                 "Lorg/gradle/internal/hash/HashCode;" +
                 "Lorg/gradle/internal/classpath/ClassPath;" +
                 ")V")
     }
+
+    private
+    val stagedProgramType = "Lorg/gradle/kotlin/dsl/execution/ExecutableProgram\$StagedProgram;"
 
     private
     fun requiresAccessors() =

--- a/subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/execution/ResidualProgramCompiler.kt
+++ b/subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/execution/ResidualProgramCompiler.kt
@@ -142,9 +142,7 @@ class ResidualProgramCompiler(
             "()Ljava/lang/String;",
             "()Ljava/lang/String;"
         ) {
-            if (secondStageScriptText.length < 16 * 1024) {
-                LDC(secondStageScriptText)
-            } else {
+            if (mightBeLargerThan64KB(secondStageScriptText)) {
                 // Large scripts are stored as a resource to overcome
                 // the 64KB string constant limitation
                 val resourcePath = storeStringToResource(secondStageScriptText)
@@ -155,10 +153,18 @@ class ResidualProgramCompiler(
                     ExecutableProgram.StagedProgram::loadScriptResource.name,
                     "(Ljava/lang/String;)Ljava/lang/String;"
                 )
+            } else {
+                LDC(secondStageScriptText)
             }
             ARETURN()
         }
     }
+
+    private
+    fun mightBeLargerThan64KB(secondStageScriptText: String) =
+        // We use a simple heuristic to avoid converting the string to bytes
+        // if all code points are in UTF32, 16KB code points would be 64KB bytes
+        secondStageScriptText.length >= 16 * 1024
 
     private
     fun storeStringToResource(secondStageScriptText: String): String {

--- a/subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/execution/ResidualProgramCompiler.kt
+++ b/subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/execution/ResidualProgramCompiler.kt
@@ -163,7 +163,7 @@ class ResidualProgramCompiler(
     private
     fun storeStringToResource(secondStageScriptText: String): String {
         val hash = Hashing.hashString(secondStageScriptText)
-        val resourcePath = "gradle-kotlin-dsl/$hash"
+        val resourcePath = "scripts/$hash.gradle.kts"
         writeResourceFile(resourcePath, secondStageScriptText)
         return resourcePath
     }

--- a/subprojects/provider/src/test/kotlin/org/gradle/kotlin/dsl/execution/ResidualProgramCompilerTest.kt
+++ b/subprojects/provider/src/test/kotlin/org/gradle/kotlin/dsl/execution/ResidualProgramCompilerTest.kt
@@ -182,8 +182,8 @@ class ResidualProgramCompilerTest : TestWithTempFiles() {
                 sourceHash,
                 null)
 
-            verify(programHost).compileSecondStageScript(
-                source.text,
+            verify(programHost).compileSecondStageOf(
+                program,
                 scriptHost,
                 stage2SettingsTemplateId,
                 sourceHash,
@@ -464,6 +464,11 @@ class ResidualProgramCompilerTest : TestWithTempFiles() {
                 program.loadSecondStageFor(programHost, scriptHost, scriptTemplateId, sourceHash, accessorsClassPath)
             }
 
+            assertThat(
+                program.secondStageScriptText,
+                equalTo(stagedProgram.source.text)
+            )
+
             inOrder(programHost) {
 
                 verify(programHost).applyPluginsTo(
@@ -479,8 +484,8 @@ class ResidualProgramCompilerTest : TestWithTempFiles() {
                     sourceHash = sourceHash,
                     accessorsClassPath = accessorsClassPath)
 
-                verify(programHost).compileSecondStageScript(
-                    stagedProgram.source.text,
+                verify(programHost).compileSecondStageOf(
+                    program,
                     scriptHost,
                     scriptTemplateId,
                     sourceHash,

--- a/subprojects/provider/src/test/kotlin/org/gradle/kotlin/dsl/integration/KotlinBuildScriptIntegrationTest.kt
+++ b/subprojects/provider/src/test/kotlin/org/gradle/kotlin/dsl/integration/KotlinBuildScriptIntegrationTest.kt
@@ -9,8 +9,40 @@ import org.hamcrest.CoreMatchers.equalTo
 import org.junit.Assert.assertThat
 import org.junit.Test
 
+import java.io.StringWriter
+
 
 class KotlinBuildScriptIntegrationTest : AbstractIntegrationTest() {
+
+    @Test
+    fun `scripts larger than 64KB are supported`() {
+
+        withBuildScriptLargerThan64KB("""
+            tasks.register("run") {
+                doLast { println("*42*") }
+            }
+        """)
+
+        assertThat(
+            build("run").output,
+            containsString("*42*")
+        )
+    }
+
+    private
+    fun withBuildScriptLargerThan64KB(suffix: String) =
+        withBuildScript(StringWriter().run {
+            var bytesWritten = 0
+            var i = 0
+            while (bytesWritten < 64 * 1024) {
+                val stmt = "val v$i = $i\n"
+                write(stmt)
+                i += 1
+                bytesWritten += stmt.length * 2
+            }
+            write(suffix)
+            toString()
+        })
 
     @Test
     fun `can use Kotlin 1 dot 3 language features`() {

--- a/subprojects/provider/src/test/kotlin/org/gradle/kotlin/dsl/integration/KotlinBuildScriptIntegrationTest.kt
+++ b/subprojects/provider/src/test/kotlin/org/gradle/kotlin/dsl/integration/KotlinBuildScriptIntegrationTest.kt
@@ -38,7 +38,7 @@ class KotlinBuildScriptIntegrationTest : AbstractIntegrationTest() {
                 val stmt = "val v$i = $i\n"
                 write(stmt)
                 i += 1
-                bytesWritten += stmt.length * 2
+                bytesWritten += stmt.toByteArray().size
             }
             write(suffix)
             toString()

